giraffe/src/utils/unzip.ts

```diff
@@ -0,0 +1,33 @@
+import {isLength} from './isLength'
+
+const isArrayLikeObject = value => {
+  return (
+    typeof value === 'object' &&
+    value !== null &&
+    typeof value !== 'function' &&
+    isLength(value.length)
+  )
+}
+
+const baseProperty = key => {
+  return object => (object == null ? undefined : object[key])
```

Comment:
```suggestion
  return object => (object === null ? undefined : object[key])
```

Reply:
Good catch. Copy-pasta failure here.

Reply:
Actually. We want the double equal here because undefined == null will result in true, which is what we want.

Reply:
TIL